src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Compatibility.Common.targets

```diff
@@ -33,6 +34,37 @@
       CompatibilitySuppressionFilePath="$(CompatibilitySuppressionFilePath)"
       BaselinePackageTargetPath="$(PackageValidationBaselinePath)"
       DisablePackageBaselineValidation="$(DisablePackageBaselineValidation)"
-      RoslynAssembliesPath="$(RoslynAssembliesPath)" />
+      RoslynAssembliesPath="$(RoslynAssembliesPath)"
+      ReferenceDirectories="@(PackageValidationReferenceDirectories)" />
+  </Target>
+
+  <PropertyGroup>
+    <_GetPackageValidationReferenceDirectoriesDependsOn Condition="'$(IsCrossTargetingBuild)' != 'true'">_GetReferenceDirectoriesForPackageValidation</_GetPackageValidationReferenceDirectoriesDependsOn>
+    <_GetPackageValidationReferenceDirectoriesDependsOn Condition="'$(IsCrossTargetingBuild)' == 'true'">_ComputeTargetFrameworkItems</_GetPackageValidationReferenceDirectoriesDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_GetReferenceDirectoriesForPackageValidation"
+          DependsOnTargets="ResolveReferences"
+          Returns="@(_ReferenceDirectoriesWithTargetFramework)">
+    <ItemGroup>
+      <_referenceDirectories Include="@(ReferencePath -> '%(RootDir)%(Directory)')" />
+      <_ReferenceDirectoriesWithTargetFramework Include="%(_referenceDirectories.Identity)"  TargetFramework="$(TargetFramework)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_GetPackageValidationReferenceDirectories"
+          DependsOnTargets="$(_GetPackageValidationReferenceDirectoriesDependsOn)"
+          Condition="'$(RunPackageValidationWithoutReferences)' != 'true'">
```

Comment:
So it looks like we have this enabled by default if I'm reading this right.  That seems OK if we are happy with the perf of this.  Might be interesting to measure that to see what it adds for a characteristic package.

Reply:
I will measure perf and report back, but so far I haven't seen a big drawback here when running these changes on dotnet/runtime so I suspect the perf hit is minimal.